Make vault client submit proof using 'confirm' messages and add fallback to other archive nodes on error

ebma opened the pull request:
- [x] Change logic about proof building from archive in accordance to #361.
- [x] Adds fallback mechanism for archive fetching. Instead of storing just one archive URL, the overlay config files now contain an array.
- [x] Changes the URL creation to not assume a trailing '/' at the end of the URL. 

Closes #361.

b-yap commented on clients/vault/src/oracle/collector/proof_builder.rs:
```diff
@@ -182,43 +182,90 @@ impl ScpMessageCollector {
+							})
+							.collect::<Vec<_>>();
+
+						let externalized_envelopes = relevant_envelopes
```
Just a suggestion:
`externalized_envelopes` is not actually used; might be better if we use `filter(...).count()` instead? That way we're not collecting duplicate envelopes

ebma replied:
Good point 👍

b-yap commented on clients/vault/src/oracle/collector/proof_builder.rs:
```diff
@@ -182,43 +182,90 @@ impl ScpMessageCollector {
+							.collect::<Vec<_>>();
+
+						// Ensure that at least one envelope is externalized
+						if externalized_envelopes.len() == 0 {
```
so in this part, it'll be `if count == 0`.

b-yap commented on clients/vault/src/oracle/collector/proof_builder.rs:
```diff
@@ -231,29 +278,33 @@ impl ScpMessageCollector {
-				tracing::debug!("Adding archived tx set for slot {}", slot);
-				let mut tx_set_map = txset_map_arc.write();
-				tx_set_map.insert(slot, target_history_entry.tx_set.clone());
+					return
```
Correct me if I'm wrong... 
let's say we have 10 archive urls. 

If at **_first_**  or **_fifth_** iteration it errors out, is this going to stop the loop and return immediately? Is there no need to check until the last url?
^ if that is the expected behaviour, please disregard this comment, :sweat:

ebma replied:
Ahh nice catch. It should actually be a `continue` and not a `return` since we of course want to continue with the other archives in case we receive an error

ebma commented:
@b-yap I addressed your feedback and added some tests specifically for the fallback. I noticed some issues while doing so and also addressed them now.

b-yap commented on clients/stellar-relay-lib/src/config.rs:
```diff
@@ -8,9 +8,9 @@ use substrate_stellar_sdk::SecretKey;
-	stellar_history_base_url: String,
-	connection_info: ConnectionInfoCfg,
-	node_info: NodeInfoCfg,
+	pub stellar_history_archive_urls: Vec<String>,
```
possibly `pub(crate)` ? I assumed this was made pubilc for testing reasons.
Not sure if we want the fields exposed outside of `stellar-relay-lib`

ebma replied:
We actually can't use `pub(crate)` here because this config lives in the `stellar-relay-lib` and the tests I added live in the `vault/oracle` crate 😅

b-yap replied:
oh, my bad. alrighty then, 👍

b-yap commented:
once the test is fixed, I  :+1:

ebma commented:
@b-yap CI passed, can you approve?
